src/olympia/users/models.py

```diff
@@ -489,13 +494,22 @@ class Meta:
 
 
 @UserProfile.on_change
-def watch_email(old_attr=None, new_attr=None, instance=None,
-                sender=None, **kw):
+def watch_changes(old_attr=None, new_attr=None, instance=None,
+                  sender=None, **kw):
     if old_attr is None:
         old_attr = {}
     if new_attr is None:
         new_attr = {}
+    # Log email changes.
     new_email, old_email = new_attr.get('email'), old_attr.get('email')
     if old_email and new_email != old_email:
         log.debug('Creating user history for user: %s' % instance.pk)
         UserHistory.objects.create(email=old_email, user_id=instance.pk)
+    # If username or display_name changes, reindex the user add-ons, if there
+    # are any.
+    if (new_attr.get('username') != old_attr.get('username') or
+            new_attr.get('display_name') != old_attr.get('display_name')):
+        from olympia.addons.tasks import index_addons
+        ids = [a.pk for a in instance.get_addons_listed()]
```

Comment:
one-character names are bad, even in list comprehensions. Let's call this `addon` maybe?